woocommerce/woocommerce-ios
# Edit Product: add inline error UI for SKU local validation error

Part of #1652 

When the user entered a SKU value that is the same as another local Product in storage, this PR replaces the error alert with an inline error UI that highlights the section and row where the error occurs.

## Changes

- Created `ErrorSectionHeaderView` and its xib for a section header view to show an error state
- Added state to `TitleAndTextFieldTableViewCell.ViewModel` with default to `normal` state
- In `ProductInventorySettingsViewController`, registered the new `ErrorSectionHeaderView` header view. Added an optional `errorTitle` to the `Section` struct, and showed the error section header in its `UITableViewDelegate` implementation
  - I learned from trials and errors that just implementing `heightForHeaderInSection` that returns `UITableView.automaticDimension` wasn't enough. `estimatedHeightForHeaderInSection` was also required, otherwise `viewForHeaderInSection` for **the first section** (only the first section) wouldn't be called 😱 
- In `ProductInventorySettingsViewController`, when receiving a SKU error:
  - Stores the error
  - Reloads the table view section view models
  - Reloads the table view
    - The error state section header is shown by `ErrorSectionHeaderView` via the section view model with `errorTitle`
    - The error state in `TitleAndTextFieldTableViewCell` is shown by checking the internal `error` property and updating its cell view model's state

## Testing

- Launch the app
- Go to Products tab
- Tap on a simple Product
- Tap "Edit"
- Tap the inventory settings row
- Enter a SKU that is used by another existing Product in the storage (already loaded on the Products tab)
- Tap "Done" --> an error state is shown
- Edit the SKU to a different value that hasn't been taken
- Tap "Done" --> the SKU is saved locally but not yet published
- Tap "Done" on the Edit Product main screen --> the SKU should be saved remotely after

## Example screenshots

@kyleaparker I made a section header view for the error above the SKU section, so we can potentially reuse this UI for an error in a section (because there is already a top padding in the grouped style table view, I made the error label with 0px top margin and 22px bottom margin)

Dark | Light
-- | --
![Simulator Screen Shot - iPhone 8 - 2020-01-03 at 20 52 08](https://user-images.githubusercontent.com/1945542/71724661-04cb4f00-2e6c-11ea-89ed-b100040fe216.png) | ![Simulator Screen Shot - iPhone 8 - 2020-01-03 at 20 52 18](https://user-images.githubusercontent.com/1945542/71724663-0563e580-2e6c-11ea-8b86-2eba23834e6f.png)


Update release notes:
